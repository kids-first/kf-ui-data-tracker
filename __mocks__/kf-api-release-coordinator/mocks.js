import {GET_STUDY_RELEASES, GET_RELEASED_STUDY} from '../../src/state/queries';
import getStudyReleases from './responses/getStudyReleases.json';
import getStudyReleasesEmpty from './responses/getStudyReleasesEmpty.json';
import allReleaseStudies from './responses/allReleaseStudies.json';

export const coordMocks = {
  getStudyReleases: {
    request: {
      query: GET_STUDY_RELEASES,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
      },
    },
    result: getStudyReleases,
  },
  getStudyReleasesError: {
    request: {
      query: GET_STUDY_RELEASES,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
      },
    },
    error: new Error('Failed to fetch releases information'),
  },
  getStudyReleasesEmpty: {
    request: {
      query: GET_STUDY_RELEASES,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
      },
    },
    result: getStudyReleasesEmpty,
  },
  allReleaseStudies: {
    request: {
      query: GET_RELEASED_STUDY,
    },
    result: allReleaseStudies,
  },
  allReleaseStudiesError: {
    request: {
      query: GET_RELEASED_STUDY,
    },
    error: new Error('Failed to fetch study releases information'),
  },
};
