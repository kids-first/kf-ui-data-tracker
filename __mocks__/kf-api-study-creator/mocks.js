import {
  ALL_STUDIES,
  GET_STUDY_BY_ID,
  MY_PROFILE,
  ALL_EVENTS,
  ALL_USERS,
  GET_PROJECTS,
  STATUS,
} from '../../src/state/queries';
import {
  CREATE_PROJECT,
  SYNC_PROJECTS,
  CREATE_STUDY,
  UPDATE_STUDY,
} from '../../src/state/mutations';
import {GET_FILE_BY_ID} from '../../src/documents/queries';
import {
  DELETE_FILE,
  UPDATE_FILE,
  UPDATE_VERSION,
  CREATE_VERSION,
  FILE_DOWNLOAD_URL,
} from '../../src/documents/mutations';
import {GET_DEV_TOKENS} from '../../src/admin/queries';
import {DELETE_DEV_TOKEN, CREATE_DEV_TOKEN} from '../../src/admin/mutations';
import allStudies from './responses/allStudies';
import allStudies_empty from './responses/allStudies_empty';
import studyByKfId from './responses/studyByKfId.json';
import studyByKfId_refetch from './responses/studyByKfId_refetch.json';
import studyByKfId_event_err from './responses/studyByKfId_event_err.json';
import studyByKfId_no_projects from './responses/studyByKfId_no_projects.json';
import studyByKfId_empty from './responses/studyByKfId_empty.json';
import deleteFile from './responses/deleteFile.json';
import fileByKfId from './responses/fileByKfId.json';
import fileByKfId_empty from './responses/fileByKfId_empty.json';
import myProfile from './responses/myProfile.json';
import updateFile from './responses/updateFile.json';
import updateVersion from './responses/updateVersion.json';
import createVersion from './responses/createVersion.json';
import createStudy from './responses/createStudy.json';
import updateStudy from './responses/updateStudy.json';
import devTokens from './responses/devTokens.json';
import createDevToken from './responses/createDevToken.json';
import allEvents from './responses/allEvents.json';
import allEvents_refetch from './responses/allEvents_refetch.json';
import allEvents_20 from './responses/allEvents_20.json';
import allEvents_40 from './responses/allEvents_40.json';
import allUsers from './responses/allUsers.json';
import allProjects from './responses/allProjects.json';
import signedUrl from './responses/signedUrl.json';
import syncProjects from './responses/syncProjects.json';
import deleteToken from './responses/deleteToken.json';
import status from './responses/status.json';
import allEvents_studyCreation from './responses/allEvents_studyCreation.json';
import allEvents_studyCreation_err from './responses/allEvents_studyCreation_err.json';
import allEvents_studyCreation_ing from './responses/allEvents_studyCreation_ing.json';

export const mocks = [
  {
    request: {
      query: ALL_STUDIES,
    },
    result: allStudies,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_8WX8QQ06',
      },
    },
    result: studyByKfId,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_8WX8QQ06',
      },
    },
    error: new Error('something went wrong'),
  },
  {
    request: {
      query: DELETE_FILE,
      variables: {
        kfId: 'SF_Y07IN1HO',
      },
    },
    result: deleteFile,
  },
  {
    request: {
      query: GET_FILE_BY_ID,
      variables: {
        kfId: 'SF_5ZPEM167',
      },
    },
    result: fileByKfId,
  },
  {
    request: {
      query: UPDATE_FILE,
      variables: {
        kfId: 'SF_5ZPEM167',
        name: 'foo bar file',
        description: 'Some description here',
        fileType: 'CLN',
      },
    },
    result: updateFile,
  },
  {
    request: {
      query: UPDATE_VERSION,
      variables: {
        versionId: 'FV_DMTKEFQW',
        description: 'version description',
        state: 'APP',
      },
    },
    result: updateVersion,
  },
  {
    request: {
      query: CREATE_VERSION,
      variables: {
        file: {
          lastModified: 1567797130510,
          name: 'organize.tsv',
          size: 7,
          type: 'text/tab-separated-values',
          webkitRelativePath: '',
        },
        fileId: 'SF_5ZPEM167',
        description: 'version description',
      },
    },
    result: createVersion,
  },
  {
    request: {
      query: MY_PROFILE,
    },
    result: myProfile,
  },
  {
    request: {
      query: GET_DEV_TOKENS,
    },
    result: devTokens,
  },
  {
    request: {
      query: CREATE_DEV_TOKEN,
      variables: {
        name: 'my token 4',
      },
    },
    result: createDevToken,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_8WX8QQ06',
        orderBy: '-created_at',
        first: 20,
      },
    },
    result: allEvents,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_8WX8QQ06',
        orderBy: '-created_at',
        first: 20,
      },
    },
    error: new Error('something went wrong with your event logs request'),
  },
  {
    request: {
      query: ALL_USERS,
    },
    result: allUsers,
  },
  {
    request: {
      query: MY_PROFILE,
    },
    error: new Error('Failed to get user profile information'),
  },
  {
    request: {
      query: GET_PROJECTS,
      variables: {
        study: '',
        deleted: false,
      },
    },
    result: allProjects,
  },
  {
    request: {
      query: GET_PROJECTS,
      variables: {
        study: '',
        deleted: false,
      },
    },
    error: new Error('Failed to get the Cavatica project data for this study'),
  },
  {
    request: {
      query: GET_PROJECTS,
      variables: {
        study: null,
        deleted: null,
      },
    },
    result: allProjects,
  },
  {
    request: {
      query: GET_PROJECTS,
      variables: {},
    },
    result: allProjects,
  },
  {
    request: {
      query: SYNC_PROJECTS,
    },
    result: syncProjects,
  },
  {
    request: {
      query: SYNC_PROJECTS,
    },
    error: new Error('Failed to sync Cavatica projects'),
  },
  {
    request: {
      query: CREATE_STUDY,
      variables: {
        workflows: ['bwa_mem', 'gatk_haplotypecaller'],
        input: {
          externalId: 'benchmark extensible e-business',
          name: 'benchmark extensible e-business',
          shortName: '',
          description: '',
          releaseDate: null,
          version: '',
          attribution: '',
          anticipatedSamples: 0,
          awardeeOrganization: '',
        },
      },
    },
    result: createStudy,
  },
  {
    request: {
      query: CREATE_STUDY,
      variables: {
        workflows: ['bwa_mem', 'gatk_haplotypecaller'],
        input: {
          externalId: 'benchmark extensible e-business',
          name: 'benchmark extensible e-business',
          shortName: '',
          description: '',
          releaseDate: null,
          version: '',
          attribution: '',
          anticipatedSamples: 0,
          awardeeOrganization: '',
        },
      },
    },
    error: new Error('Failed to create the new study'),
  },
  {
    request: {
      query: UPDATE_STUDY,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
        input: {
          externalId: 'test_study_external_id',
          name: 'benchmark extensible e-business',
          shortName: '',
          description:
            'Sudy description in markdown, commonly the X01 abstract text. --  As test study description',
          releaseDate: '2019-08-24',
          anticipatedSamples: 100,
          awardeeOrganization: 'CHOP',
          attribution: '',
          version: '',
          bucket: 'stay-wrong-ready',
        },
      },
    },
    result: updateStudy,
  },
  {
    request: {
      query: UPDATE_STUDY,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
        input: {
          externalId: 'test_study_external_id',
          name: 'benchmark extensible e-business',
          shortName: '',
          description:
            'Sudy description in markdown, commonly the X01 abstract text. --  As test study description',
          releaseDate: '2019-08-24',
          anticipatedSamples: 100,
          awardeeOrganization: 'CHOP',
          attribution: '',
          version: '',
          bucket: 'stay-wrong-ready',
        },
      },
    },
    error: new Error('Failed to update the study'),
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        first: 20,
      },
    },
    result: allEvents,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_8WX8QQ06',
      },
    },
    result: studyByKfId_refetch,
  },
  {
    request: {
      query: CREATE_VERSION,
      variables: {
        file: {},
        fileId: 'SF_5ZPEM167',
        description: 'my changes',
      },
    },
    result: createVersion,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        username: 'Justin Heath',
        first: 20,
      },
    },
    result: allEvents_refetch,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        username: 'Justin Heath',
        studyId: 'SD_SG5N41K8',
        first: 20,
      },
    },
    result: allEvents_refetch,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        username: 'Justin Heath',
        studyId: 'SD_SG5N41K8',
        eventType: 'FV_CRE',
        first: 20,
      },
    },
    result: allEvents_refetch,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        first: 20,
      },
    },
    error: new Error('Failed to fetch events information'),
  },
  {
    request: {
      query: FILE_DOWNLOAD_URL,
      variables: {
        studyId: 'SD_00000000',
        fileId: 'SF_5ZPEM167',
        versionId: 'FV_TT8NF09R',
      },
    },
    result: signedUrl,
  },
  {
    request: {
      query: GET_PROJECTS,
      variables: {},
    },
    error: new Error('Failed to fetch projects information'),
  },
  {
    request: {
      query: DELETE_DEV_TOKEN,
      variables: {
        name: 'my token 1',
      },
    },
    result: deleteToken,
  },
  {
    request: {
      query: CREATE_DEV_TOKEN,
      variables: {
        name: 'my token 4',
      },
    },
    error: new Error('Failed to create new token'),
  },
  {
    request: {
      query: GET_DEV_TOKENS,
    },
    error: new Error('Failed to fetch tokens information'),
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        studyId: 'SD_8WX8QQ06',
        eventType: 'FV_CRE',
        first: 20,
      },
    },
    result: allEvents_refetch,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        studyId: 'SD_8WX8QQ06',
      },
    },
    result: allEvents_studyCreation,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        studyId: 'SD_8WX8QQ06',
      },
    },
    result: allEvents_studyCreation_err,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        studyId: 'SD_8WX8QQ06',
      },
    },
    result: allEvents_studyCreation_ing,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_8WX8QQ06',
        orderBy: '-created_at',
      },
    },
    error: new Error('Failed to fetch events information'),
  },
  {
    request: {
      query: STATUS,
    },
    result: status,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_8WX8QQ06',
      },
    },
    result: studyByKfId_event_err,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_8WX8QQ06',
      },
    },
    result: studyByKfId_no_projects,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {orderBy: '-created_at', first: 20},
    },
    result: allEvents_20,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        first: 20,
        cursor: 'YXJyYXljb25uZWN0aW9uOjE5',
      },
    },
    result: allEvents_40,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_8WX8QQ06',
        orderBy: '-created_at',
        first: 20,
      },
    },
    result: allEvents_20,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_8WX8QQ06',
        orderBy: '-created_at',
        first: 20,
        cursor: 'YXJyYXljb25uZWN0aW9uOjE5',
      },
    },
    result: allEvents_40,
  },
  {
    request: {
      query: ALL_STUDIES,
    },
    error: new Error('Failed to fetch study information'),
  },
  {
    request: {
      query: ALL_STUDIES,
    },
    result: allStudies_empty,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_INVALID',
      },
    },
    result: studyByKfId_empty,
  },
  {
    request: {
      query: GET_FILE_BY_ID,
      variables: {
        kfId: 'SF_INVALIDS',
      },
    },
    result: fileByKfId_empty,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_INVALID',
        orderBy: '-created_at',
        first: 20,
      },
    },
    result: allEvents,
  },
  {
    request: {
      query: CREATE_STUDY,
      variables: {
        input: {
          name: 'benchmark extensible e-business',
          shortName: '',
          description: '',
          externalId: 'benchmark-extensible-e-business',
          collaborators: ['VXNlck5vZGU6MQ==', 'VXNlck5vZGU6Me=='],
        },
        workflows: [],
      },
    },
    result: createStudy,
  },
  {
    request: {
      query: CREATE_STUDY,
      variables: {
        workflows: [],
        input: {
          externalId: 'benchmark-extensible-e-business',
          name: 'benchmark extensible e-business',
          shortName: '',
          description: '',
          collaborators: [],
        },
      },
    },
    error: new Error('Failed to create the new research study'),
  },
  {
    request: {
      query: UPDATE_STUDY,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
        input: {
          name: 'benchmark extensible e-business',
          shortName: 'Test short name input on update',
          description:
            'Sudy description in markdown, commonly the X01 abstract text. --  As test study description',
          collaborators: ['VXNlck5vZGU6MQ==', 'VXNlck5vZGU6Me=='],
        },
      },
    },
    result: updateStudy,
  },
  {
    request: {
      query: UPDATE_STUDY,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
        input: {
          name: 'benchmark extensible e-business',
          shortName: '',
          description:
            'Sudy description in markdown, commonly the X01 abstract text. --  As test study description',
          collaborators: ['VXNlck5vZGU6MQ==', 'VXNlck5vZGU6Me=='],
        },
      },
    },
    error: new Error('Failed to update the study'),
  },
  {
    request: {
      query: ALL_USERS,
    },
    error: new Error('Failed to fetch users information'),
  },
];
