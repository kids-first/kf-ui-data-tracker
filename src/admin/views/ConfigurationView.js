import React from 'react';
import {Helmet} from 'react-helmet';
import {useQuery} from '@apollo/react-hooks';
import {Container, Header, Segment, Icon, Table} from 'semantic-ui-react';
import TimeAgo from 'react-timeago';
import {STATUS} from '../../state/queries';

const FeatureTable = ({features}) => (
  <Table
    tableData={features}
    headerRow={['Feature', {content: 'Enabled', textAlign: 'center'}]}
    renderBodyRow={(data, index) => ({
      cells: [
        {content: data.name},
        {
          content: <Icon name={data.enabled ? 'check' : 'delete'} />,
          textAlign: 'center',
        },
      ],
      warning: !data.enabled,
      positive: data.enabled,
    })}
  />
);

const SettingsTable = ({settings}) => (
  <Table
    tableData={settings}
    headerRow={['Setting', {content: 'Value', textAlign: 'left'}]}
    renderBodyRow={(data, index) => ({
      cells: [
        {content: data.name},
        {
          content: data.value,
          textAlign: 'left',
        },
      ],
    })}
  />
);

const Queues = ({queues}) => (
  <Table
    tableData={queues}
    headerRow={[
      'Queue',
      'Jobs',
      'Workers',
      {content: 'Started Jobs', textAlign: 'right'},
      {content: 'Failed Jobs', textAlign: 'right'},
      {content: 'Finished Jobs', textAlign: 'right'},
    ]}
    renderBodyRow={(data, index) => ({
      cells: [
        {content: data.name},
        {content: data.jobs},
        {content: data.workers},
        {content: data.started_jobs, textAlign: 'right'},
        {content: data.failed_jobs, textAlign: 'right'},
        {content: data.finished_jobs, textAlign: 'right'},
      ],
    })}
  />
);

const Jobs = ({jobs}) => (
  <Table
    tableData={jobs}
    headerRow={[
      'Job',
      {content: 'Status', textAlign: 'center'},
      {content: 'Error Message'},
      {content: 'Last Run', textAlign: 'right'},
      {content: 'Next Run', textAlign: 'right'},
    ]}
    renderBodyRow={(data, index) => ({
      cells: [
        {content: data.name},
        {
          content: data.failing ? (
            <Icon name="delete" />
          ) : data.active ? (
            <Icon name="check" />
          ) : (
            <Icon name="delete" />
          ),
          textAlign: 'center',
        },
        {content: data.lastError},
        {
          content: <TimeAgo date={data.lastRun} live={false} />,
          textAlign: 'right',
        },
        {
          content: <TimeAgo date={data.enqueuedAt} live={false} />,
          textAlign: 'right',
        },
      ],
      error: data.failing,
    })}
  />
);

const ConfigurationView = () => {
  const {data} = useQuery(STATUS);

  const features =
    data &&
    data.status.features &&
    Object.keys(data.status.features)
      .filter(key => !key.includes('__'))
      .map(key => ({
        name: key,
        enabled: data.status.features[key],
      }));

  const settings =
    data &&
    data.status.settings &&
    Object.keys(data.status.settings)
      .filter(key => !key.includes('__'))
      .map(key => ({
        name: key,
        value: data.status.settings[key],
      }));

  const queues = data && data.status.queues && JSON.parse(data.status.queues);

  const jobs =
    data &&
    data.status.jobs &&
    data.status.jobs.edges.map(({node}) => ({
      name: node.name,
      active: node.active,
      failing: node.failing,
      lastRun: node.lastRun,
      lastError: node.lastError,
      enqueuedAt: node.enqueuedAt,
    }));

  return (
    <>
      <Helmet>
        <title>KF Data Tracker - Configuration</title>
      </Helmet>
      <Container as={Segment} basic>
        <Header as="h3">Data Tracker Configuration</Header>
        <Segment basic>
          These are the features and settings that the Data Tracker is running
          with. They are for debug purposes only and require deployment changes
          to be modified.
        </Segment>

        <Header as="h4">Feature Flags</Header>
        <FeatureTable features={features} />

        <Header as="h4">Settings</Header>
        <SettingsTable settings={settings} />

        <Header as="h4">Queues</Header>
        <Queues queues={queues} />

        <Header as="h4">Jobs</Header>
        <Jobs jobs={jobs} />
      </Container>
    </>
  );
};

export default ConfigurationView;
