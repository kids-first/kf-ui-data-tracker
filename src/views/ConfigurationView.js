import React from 'react';
import {Helmet} from 'react-helmet';
import {useQuery} from '@apollo/react-hooks';
import {Container, Header, Segment, Icon, Table} from 'semantic-ui-react';
import {STATUS} from '../state/queries';

const FeatureTable = ({features}) => (
  <Table
    tableData={features}
    headerRow={['Feature', {content: 'Enabled', textAlign: 'center'}]}
    renderBodyRow={(data, index) => ({
      cells: [
        {content: data.name},
        {
          content: <Icon name={data.enabled ? 'check' : 'delete'} />,
          textAlign: 'center',
        },
      ],
      warning: !data.enabled,
      positive: data.enabled,
    })}
  />
);

const SettingsTable = ({settings}) => (
  <Table
    tableData={settings}
    headerRow={['Setting', {content: 'Value', textAlign: 'left'}]}
    renderBodyRow={(data, index) => ({
      cells: [
        {content: data.name},
        {
          content: data.value,
          textAlign: 'left',
        },
      ],
    })}
  />
);

const Queues = ({queues}) => (
  <Table
    tableData={queues}
    headerRow={[
      'Queue',
      'Jobs',
      'Workers',
      {content: 'Started Jobs', textAlign: 'right'},
      {content: 'Failed Jobs', textAlign: 'right'},
      {content: 'Finished Jobs', textAlign: 'right'},
    ]}
    renderBodyRow={(data, index) => ({
      cells: [
        {content: data.name},
        {content: data.jobs},
        {content: data.workers},
        {content: data.started_jobs, textAlign: 'right'},
        {content: data.failed_jobs, textAlign: 'right'},
        {content: data.finished_jobs, textAlign: 'right'},
      ],
    })}
  />
);

const ConfigurationView = () => {
  const {data} = useQuery(STATUS);

  const features =
    data &&
    data.status.features &&
    Object.keys(data.status.features)
      .filter(key => !key.includes('__'))
      .map(key => ({
        name: key,
        enabled: data.status.features[key],
      }));

  const settings =
    data &&
    data.status.settings &&
    Object.keys(data.status.settings)
      .filter(key => !key.includes('__'))
      .map(key => ({
        name: key,
        value: data.status.settings[key],
      }));

  const queues = data && data.status.queues && JSON.parse(data.status.queues);

  // const queues = [
  //   {
  //     name: 'cavatica',
  //     jobs: 0,
  //     workers: 0,
  //     finished_jobs: 100,
  //   },
  //   {
  //     name: 'default',
  //     jobs: 0,
  //     workers: 0,
  //     finished_jobs: 100,
  //   },
  // ];

  return (
    <>
      <Helmet>
        <title>KF Data Tracker - Configuration</title>
      </Helmet>
      <Container as={Segment} basic>
        <Header as="h3">Data Tracker Configuration</Header>
        <Segment basic>
          These are the features and settings that the Data Tracker is running
          with. They are for debug purposes only and require deployment changes
          to be modified.
        </Segment>

        <Header as="h4">Feature Flags</Header>
        <FeatureTable features={features} />

        <Header as="h4">Settings</Header>
        <SettingsTable settings={settings} />

        <Header as="h4">Queues</Header>
        <Queues queues={queues} />
      </Container>
    </>
  );
};

export default ConfigurationView;
