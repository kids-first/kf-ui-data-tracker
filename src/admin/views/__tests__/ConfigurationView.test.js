import React from 'react';
import wait from 'waait';
import {render} from '@testing-library/react';
import {MockedProvider} from '@apollo/react-testing';
import {mocks} from '../../../../__mocks__/kf-api-study-creator/mocks';
import ConfigurationView from '../ConfigurationView';

jest.mock('auth0-js');

it('renders the configuration page', async () => {
  const tree = render(
    <MockedProvider mocks={mocks}>
      <ConfigurationView />
    </MockedProvider>,
  );
  await wait(10);

  expect(tree.container).toMatchSnapshot();
});
