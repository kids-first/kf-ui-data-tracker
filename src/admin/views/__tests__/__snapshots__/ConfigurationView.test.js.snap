// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the configuration page 1`] = `
<div>
  <div
    class="ui basic segment ui container"
  >
    <h3
      class="ui header"
    >
      Data Tracker Configuration
    </h3>
    <div
      class="ui basic segment"
    >
      These are the features and settings that the Data Tracker is running with. They are for debug purposes only and require deployment changes to be modified.
    </div>
    <h4
      class="ui header"
    >
      Feature Flags
    </h4>
    <table
      class="ui table"
    >
      <thead
        class=""
      >
        <tr
          class=""
        >
          <th
            class=""
          >
            Feature
          </th>
          <th
            class="center aligned"
          >
            Enabled
          </th>
        </tr>
      </thead>
      <tbody
        class=""
      >
        <tr
          class="positive"
        >
          <td
            class=""
          >
            studyCreation
          </td>
          <td
            class="center aligned"
          >
            <i
              aria-hidden="true"
              class="check icon"
            />
          </td>
        </tr>
        <tr
          class="positive"
        >
          <td
            class=""
          >
            studyUpdates
          </td>
          <td
            class="center aligned"
          >
            <i
              aria-hidden="true"
              class="check icon"
            />
          </td>
        </tr>
        <tr
          class="positive"
        >
          <td
            class=""
          >
            cavaticaCreateProjects
          </td>
          <td
            class="center aligned"
          >
            <i
              aria-hidden="true"
              class="check icon"
            />
          </td>
        </tr>
        <tr
          class="positive"
        >
          <td
            class=""
          >
            cavaticaCopyUsers
          </td>
          <td
            class="center aligned"
          >
            <i
              aria-hidden="true"
              class="check icon"
            />
          </td>
        </tr>
        <tr
          class="warning"
        >
          <td
            class=""
          >
            cavaticaMountVolumes
          </td>
          <td
            class="center aligned"
          >
            <i
              aria-hidden="true"
              class="delete icon"
            />
          </td>
        </tr>
        <tr
          class="warning"
        >
          <td
            class=""
          >
            bucketserviceCreateBuckets
          </td>
          <td
            class="center aligned"
          >
            <i
              aria-hidden="true"
              class="delete icon"
            />
          </td>
        </tr>
      </tbody>
    </table>
    <h4
      class="ui header"
    >
      Settings
    </h4>
    <table
      class="ui table"
    >
      <thead
        class=""
      >
        <tr
          class=""
        >
          <th
            class=""
          >
            Setting
          </th>
          <th
            class="left aligned"
          >
            Value
          </th>
        </tr>
      </thead>
      <tbody
        class=""
      >
        <tr
          class=""
        >
          <td
            class=""
          >
            dataserviceUrl
          </td>
          <td
            class="left aligned"
          >
            http://dataservice
          </td>
        </tr>
        <tr
          class=""
        >
          <td
            class=""
          >
            bucketserviceUrl
          </td>
          <td
            class="left aligned"
          >
            http://bucketservice
          </td>
        </tr>
        <tr
          class=""
        >
          <td
            class=""
          >
            cavaticaUrl
          </td>
          <td
            class="left aligned"
          >
            https://cavatica-api.sbgenomics.com/v2
          </td>
        </tr>
        <tr
          class=""
        >
          <td
            class=""
          >
            cavaticaDeliveryAccount
          </td>
          <td
            class="left aligned"
          >
            kids-first-dev
          </td>
        </tr>
        <tr
          class=""
        >
          <td
            class=""
          >
            cavaticaHarmonizationAccount
          </td>
          <td
            class="left aligned"
          />
        </tr>
        <tr
          class=""
        >
          <td
            class=""
          >
            cavaticaUserAccessProject
          </td>
          <td
            class="left aligned"
          >
            kids-first-drc/user-access
          </td>
        </tr>
      </tbody>
    </table>
    <h4
      class="ui header"
    >
      Queues
    </h4>
    <table
      class="ui table"
    >
      <thead
        class=""
      >
        <tr
          class=""
        >
          <th
            class=""
          >
            Queue
          </th>
          <th
            class=""
          >
            Jobs
          </th>
          <th
            class=""
          >
            Workers
          </th>
          <th
            class="right aligned"
          >
            Started Jobs
          </th>
          <th
            class="right aligned"
          >
            Failed Jobs
          </th>
          <th
            class="right aligned"
          >
            Finished Jobs
          </th>
        </tr>
      </thead>
      <tbody
        class=""
      >
        <tr
          class=""
        >
          <td
            class=""
          >
            cavatica
          </td>
          <td
            class=""
          >
            0
          </td>
          <td
            class=""
          >
            0
          </td>
          <td
            class="right aligned"
          >
            0
          </td>
          <td
            class="right aligned"
          >
            0
          </td>
          <td
            class="right aligned"
          >
            1
          </td>
        </tr>
        <tr
          class=""
        >
          <td
            class=""
          >
            default
          </td>
          <td
            class=""
          >
            0
          </td>
          <td
            class=""
          >
            0
          </td>
          <td
            class="right aligned"
          >
            0
          </td>
          <td
            class="right aligned"
          >
            0
          </td>
          <td
            class="right aligned"
          >
            0
          </td>
        </tr>
      </tbody>
    </table>
    <h4
      class="ui header"
    >
      Jobs
    </h4>
    <table
      class="ui table"
    >
      <thead
        class=""
      >
        <tr
          class=""
        >
          <th
            class=""
          >
            Job
          </th>
          <th
            class="center aligned"
          >
            Status
          </th>
          <th
            class=""
          >
            Error Message
          </th>
          <th
            class="right aligned"
          >
            Last Run
          </th>
          <th
            class="right aligned"
          >
            Next Run
          </th>
        </tr>
      </thead>
      <tbody
        class=""
      >
        <tr
          class=""
        >
          <td
            class=""
          >
            cavatica_sync
          </td>
          <td
            class="center aligned"
          >
            <i
              aria-hidden="true"
              class="check icon"
            />
          </td>
          <td
            class=""
          >
            
          </td>
          <td
            class="right aligned"
          >
            <time
              datetime="2020-02-10T19:16:43.821Z"
              title="2020-02-10T19:16:43.821120+00:00"
            >
              10 months from now
            </time>
          </td>
          <td
            class="right aligned"
          >
            <time
              datetime="2020-02-10T19:21:37.000Z"
              title="2020-02-10T19:21:37+00:00"
            >
              10 months from now
            </time>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
